diff --git a/src/pstree.c b/src/pstree.c
index 53b8f8c..b7270f2 100644
--- a/src/pstree.c
+++ b/src/pstree.c
@@ -335,8 +335,12 @@ add_proc (const char *comm, pid_t pid, pid_t ppid, uid_t uid,
 #else  /*WITH_SELINUX*/
     parent = new_proc ("?", ppid, 0);
 #endif /*WITH_SELINUX*/
-  add_child (parent, this);
-  this->parent = parent;
+
+  if (pid != 0)
+  {
+    add_child (parent, this);
+    this->parent = parent;
+  }
 }
 
