diff -up ./doc/peekfd.1.ori ./doc/peekfd.1
--- ./doc/peekfd.1.ori	2016-12-13 19:58:35.230619102 +0100
+++ ./doc/peekfd.1	2016-12-13 19:58:52.300545794 +0100
@@ -8,7 +8,7 @@ peekfd \- peek at file descriptors of ru
 .B peekfd 
 .RB [ \-8 , \-\-eight\-bit\-clean ]
 .RB [ \-n , \-\-no\-headers ]
-.RB [ \-f , \-\-follow ]
+.RB [ \-c , \-\-follow ]
 .RB [ \-d , \-\-duplicates\-removed ]
 .RB [ \-V , \-\-version ]
 .RB [ \-h , \-\-help ]
@@ -46,7 +46,7 @@ The following diagnostics may be issued
 .B 
 Error attaching to pid ...
 .RS
-An unknown error occured while attempted to attach to a process.. you may need to be root.
+An unknown error occurred while attempted to attach to a process.. you may need to be root.
 .SH BUGS
 Probably lots.  Don't be surprised if the process you are monitoring dies.
 .SH AUTHOR
diff -up ./src/peekfd.c.ori ./src/peekfd.c
--- ./src/peekfd.c.ori	2016-12-13 19:58:46.246571793 +0100
+++ ./src/peekfd.c	2016-12-13 19:58:52.300545794 +0100
@@ -93,12 +93,12 @@ void print_version()
 void usage() {
 	fprintf(stderr, _(
       "Usage: peekfd [-8] [-n] [-c] [-d] [-V] [-h] <pid> [<fd> ..]\n"
-	  "    -8 output 8 bit clean streams.\n"
-	  "    -n don't display read/write from fd headers.\n"
-	  "    -c peek at any new child processes too.\n"
-	  "    -d remove duplicate read/writes from the output.\n"
-	  "    -V prints version info.\n"
-	  "    -h prints this help.\n"
+	  "    -8, --eight-bit-clean        output 8 bit clean streams.\n"
+	  "    -n, --no-headers             don't display read/write from fd headers.\n"
+	  "    -c, --follow                 peek at any new child processes too.\n"
+	  "    -d, --duplicates-removed     remove duplicate read/writes from the output.\n"
+	  "    -V, --version                prints version info.\n"
+	  "    -h, --help                   prints this help.\n"
 	  "\n"
 	  "  Press CTRL-C to end output.\n"));
 }
@@ -126,7 +126,7 @@ int main(int argc, char **argv)
     struct option options[] = {
       {"eight-bit-clean", 0, NULL, '8'},
       {"no-headers", 0, NULL, 'n'},
-      {"follow", 0, NULL, 'f'},
+      {"follow", 0, NULL, 'c'},
       {"duplicates-removed", 0, NULL, 'd'},
       {"help", 0, NULL, 'h'},
       {"version", 0, NULL, 'V'},
