diff -Naur psmisc-22.20.orig/src/fuser.c psmisc-22.20/src/fuser.c
--- psmisc-22.20.orig/src/fuser.c	2012-09-19 12:54:03.000000000 +0200
+++ psmisc-22.20/src/fuser.c	2015-07-31 19:55:14.000000000 +0200
@@ -1900,7 +1900,7 @@
 
 static void init_mntinfo(void)
 {
-	char mpoint[PATH_MAX + 1];
+	char mpoint[PATH_MAX*4 + 1]; // octal escaping takes 4 chars per 1 char
 	int mid, parid, max = 0;
 	uint maj, min;
 	list_t sort;
