diff -Naur psmisc-22.6.orig/src/fuser.c psmisc-22.6/src/fuser.c
--- psmisc-22.6.orig/src/fuser.c	2014-01-06 18:03:43.000000000 +0100
+++ psmisc-22.6/src/fuser.c	2014-01-06 18:05:15.674298255 +0100
@@ -917,7 +917,7 @@
     					else
     						fprintf(stderr, " %-8s ", pwent->pw_name);
     				}
-    				printf("%6d", pptr->pid);
+    				printf(" %5d", pptr->pid);
     				fflush(stdout);
     				if (opts & OPT_VERBOSE) {
     					fprintf(stderr, " %c%c%c%c%c ",
